Tests/PluginGeneratorTests/RulesPluginBuilderTests.cs

```diff
@@ -38,25 +38,143 @@ public void RulesPluginBuilder_CreateSimplePlugin_Succeeds()
                 Developers = typeof(RulesPluginBuilder).FullName
             };
 
-            RulesPluginBuilder builder = new RulesPluginBuilder(jdkWrapper, new TestLogger());
```

Comment:
Previously, this test was an integration test rather than a unit test: it actually did the compilation and built the jar.
That wasn't ideal and it's worse now that we're calling out to Maven Central to get jars on the fly. However, the recent refactoring means we can now easily inject mocks so the test doesn't need the JDK and doesn't call out to Maven Central i.e. it is just a unit test now.

The existing Roslyn plugin generator integration test provides test coverage of an end to end build.